x-pack/plugins/endpoint/public/plugin.ts

```diff
@@ -9,12 +9,14 @@ import { EmbeddableSetup } from 'src/plugins/embeddable/public';
 import { DataPublicPluginStart } from 'src/plugins/data/public';
 import { i18n } from '@kbn/i18n';
 import { ResolverEmbeddableFactory } from './embeddables/resolver';
+import { IngestManagerSetup } from '../../ingest_manager/public';
 
 export type EndpointPluginStart = void;
 export type EndpointPluginSetup = void;
 export interface EndpointPluginSetupDependencies {
   embeddable: EmbeddableSetup;
   data: DataPublicPluginStart;
+  ingestManager: IngestManagerSetup;
 }
 export interface EndpointPluginStartDependencies {
   data: DataPublicPluginStart;
```

Comment:
Not sure about this, but I think you need to add `ingestManager` dependency here as well, and you should then access it from `depsStart` which are provided during `mount()`. Similar to what the use of the `data` plugin - you can then retrieve it from the StartDependency object and should not have the need to pass `ingestManager` down to the components in `index.tsx` since `depsStart` is already passed in.

The way it is set now, you are hlding on to a `setup` value and using it during start/mount. Not sure if thats ok or not.

Reply:
Gotcha, I'll give that a shot.